pygls/protocol.py

```diff
@@ -208,6 +214,10 @@ def default_converter():
 
     converter.register_structure_hook(JsonRPCNotification, _params_field_structure_hook)
 
+    converter.register_structure_hook(
```

Comment:
@alcarney I was surprised that I needed to add this here. I feel like I must be doing something wrong or misunderstanding something? Does it make sense to you?

Reply:
I'd say it's almost expected, we're probably the first project to be exercising parts of lsprotocol's code base so it's likely we'd run into a missing structure hook here and there. (e.g. microsoft/lsprotocol#259)

Probably worth opening a PR/issue and getting it fixed upstream :)

Reply:
Ohh, thanks Alex! I've added an issue over at https://github.com/microsoft/lsprotocol/issues/270